### PR TITLE
Add analytics event for combining/extracting columns

### LINKED ISSUE #41766: Add analytics events for combine shortcut in notebook editor
Add an analytics event for the [combine shortcut](https://www.notion.so/metabase/Extract-column-shortcut-is-used-in-the-notebook-editor-88d7f16b66ef49528eda6516036506cd).

### PR DESCRIPTION
**Pull request by romeovs**
Closes https://github.com/metabase/metabase/issues/41766

[Diffs for the schema change](https://github.com/metabase/metabase/compare/37474ff170d29d33f26d166b52327443377558c2..a8860f188b7f77b1baa21eefb108a888f0350007#diff-64d1eaece53349310a4aebe48ef3b84eca57a5c4f96b265daabda6aee08ffaef)

### Description

Adds an analytics events that fires when the custom expression for combining or extracting columns is inserted via the notebook shortcut, or via the chill mode column header.

- `column_combine_via_shortcut`
- `column_combine_via_column_header`
- `column_extract_via_shortcut`
- `column_extract_via_column_header`

Adds an new property `custom_expressions_used` to track the expressions used.

I chose to add all of these at once here to avoid having to create new schema versions in 4 PR's.

### CODE REVIEW

File: frontend/src/metabase/query_builder/components/expressions/analytics.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { trackSchemaEvent } from "metabase/lib/analytics";
+
+export const trackColumnCombineViaShortcut = () => {
+  trackSchemaEvent("question", "1-0-4", {
+    event: "column_combine_via_shortcut",
+    custom_expressions_used: ["concat"],
+  });
```
Comment (Somtom) [inline frontend/src/metabase/query_builder/components/expressions/analytics.ts:7]: **suggestion/question (not-blocking):**

Does it make sense to populate other properties of the event as well here?
It seems like the following could make sense (if we have the info available):
- `visualization_type`
- `database_id`
- ... I guess we don't have information about the `question_id` or something like that, do we?

**Review by Somtom — COMMENTED** [review]
Looks good to me. Just a question regarding the other properties in the question event schema 🙂

File: frontend/src/metabase/query_builder/components/expressions/analytics.ts (hunk 1 above)
Comment (romeovs) [reply frontend/src/metabase/query_builder/components/expressions/analytics.ts:7]: I think `database_id` is the only one that is trivial to find and pass. Is that fine?

Comment (Somtom) [reply frontend/src/metabase/query_builder/components/expressions/analytics.ts]: Yeah sure - let's start with that one 🙂 .

Just to check: I had the assumption that the `column_combine_via_shortcut` would only happen in table vizualizations but that seems not to hold, does it?

File: frontend/src/metabase/query_builder/components/expressions/analytics.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { trackSchemaEvent } from "metabase/lib/analytics";
+import * as Lib from "metabase-lib";
+
+export const trackColumnCombineViaShortcut = (query: Lib.Query) => {
```
Comment (ranquild) [inline frontend/src/metabase/query_builder/components/expressions/analytics.ts:4]: This should probably go to `metabase/query_builder` folder

File: snowplow/iglu-client-embedded/schemas/com.metabase/question/jsonschema/1-0-4 [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+      "type": [
+        "integer",
+        "null"
+      ],
+      "minimum": 0,
+      "maximum": 2147483647
+    },
+    "custom_expressions_used": {
+      "description": "A list of names of expression functions and aggregations used",
+      "type": "array",
```
Comment (ranquild) [inline snowplow/iglu-client-embedded/schemas/com.metabase/question/jsonschema/1-0-4:77]: It should allow `null` as well for events that haven't this property:
```
"type": [
        "array",
        "null"
      ]
```

File: frontend/src/metabase/query_builder/components/expressions/analytics.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+  trackSchemaEvent("question", "1-0-4", {
```
Comment (ranquild) [inline frontend/src/metabase/query_builder/components/expressions/analytics.ts:5]: `question_id` is required and cannot be omitted unfortunately. You would also have to default to `0` for unsaved questions.

File: frontend/src/metabase/query_builder/components/expressions/analytics.ts (hunk 2 above)
Comment (ranquild) [inline frontend/src/metabase/query_builder/components/expressions/analytics.ts:4]: Make sure to add e2e tests for this event. Search for `describeWithSnowplow` for examples.

**Review by ranquild — CHANGES_REQUESTED** [review]
Inline comments

File: frontend/src/metabase/query_builder/analytics.js [hunk 5]
```diff
@@ -31,3 +32,12 @@ export const trackNotebookNativePreviewShown = (question, isShown) => {
     question_id: question.id() ?? 0,
   });
 };
+
+export const trackColumnCombineViaShortcut = (query, question) => {
```
Comment (ranquild) [inline frontend/src/metabase/query_builder/analytics.js:36]: If you pass `question` you can use `question.query()` to know to pass `query`

**Review by ranquild — APPROVED** [review]

**Pull request merged**